static/app/components/showHidePanel.tsx

```diff
@@ -0,0 +1,59 @@
+import {ReactNode, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {Button} from 'sentry/components/button';
+import {IconChevron} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+
+interface Props {
+  content: () => ReactNode;
+  expandedByDefault: boolean;
+  headerHide: () => ReactNode;
```

Comment:
Why do you need functions that return react nodes here?

An alternative to this is to leverage composition and pass children directly. That way your component usage would look something like

```
<ShowHidePanel expandedByDefault={true/false}>
  <SomeContent/>
</ShowHidePanel>
```